File: Examples/5/BcastRecCompBcast/src/course/examples/BroadcastReceiver/CompoundBroadcast/Receiver3.java
package course.examples.BroadcastReceiver.CompoundBroadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Vibrator;
import android.util.Log;
import android.widget.Toast;

public class Receiver3 extends BroadcastReceiver {
	private final String TAG = "Receiver3";

	@Override
	public void onReceive(Context context, Intent intent) {
		Log.i(TAG, "INTENT RECEIVED");

		Vibrator v = (Vibrator) context
				.getSystemService(Context.VIBRATOR_SERVICE);
		v.vibrate(500);

		Toast.makeText(context, "INTENT RECEIVED by Receiver3", Toast.LENGTH_LONG).show();
	}

}

File: ExamplesKotlin/BcastRecCompBcast/app/src/main/java/course/examples/broadcastreceiver/compoundbroadcast/Receiver3.kt
package course.examples.broadcastreceiver.compoundbroadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.VibrationEffect
import android.os.Vibrator
import android.util.Log
import android.widget.Toast

class Receiver3 : BroadcastReceiver() {

    companion object {

        private const val TAG = "Receiver3"
    }

    override fun onReceive(context: Context, intent: Intent) {
        Log.i(TAG, "INTENT RECEIVED")
    }

}
